### PR TITLE
Add jukaie/jk01 keyboard 

### PR DESCRIPTION
**Pull request by jukaie**
## Description

Add jukaie/jk01 keyboard.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/jukaie/jk01/config.h [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+/* COL2ROW or ROW2COL */
+#define DIODE_DIRECTION COL2ROW
+
+/* Hold ESC on start up to clear EEPROM and boot into bootloader mode */
+#define BOOTMAGIC_LITE_ROW 1
+#define BOOTMAGIC_LITE_COLUMN 3
+
+#define TAP_CODE_DELAY 10
+
```
Comment (waffle87) [inline keyboards/jukaie/jk01/config.h:37]: This should all be migrated to `info.json`. You can use `qmk info -f json -kb jukaie/jk01` to help.
```suggestion
```

File: keyboards/jukaie/jk01/jk01.c [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+},
+{
+    4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4,
+    4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4,
+    4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4,
+    8, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4,
+    4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4,
+    4, 4, 4, 4, 4, 4, 4, 4, 4, 4,
+}
+};
```
Comment (waffle87) [inline keyboards/jukaie/jk01/jk01.c:243]: This can also be migrated to `info.json`.

File: keyboards/jukaie/jk01/jk01.h [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+#pragma once
+
+#include "quantum.h"
+
+#define ___ KC_NO
+
+
+
+#define LAYOUT( \
```
Comment (waffle87) [inline keyboards/jukaie/jk01/jk01.h:25]: This should also be migrated to `info.json`, and this file removed.

File: keyboards/jukaie/jk01/jk01.c [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "jk01.h"
```
Comment (waffle87) [inline keyboards/jukaie/jk01/jk01.c:17]:
```suggestion
#include "quantum.h"
```

File: keyboards/jukaie/jk01/keymaps/default/readme.md [hunk 5]
```diff
@@ -0,0 +1 @@
+# The Default JK01 Layout
```
Comment (waffle87) [inline keyboards/jukaie/jk01/keymaps/default/readme.md:1]: This file should be removed.

File: keyboards/jukaie/jk01/keymaps/via/keymap.c [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+
+
+// Each layer gets a name for readability, which is then used in the keymap matrix below.
+// The underscores don't mean anything - you can have a layer called STUFF or any other name.
+enum my_keycodes {
+    KC_MC = SAFE_RANGE,
```
Comment (waffle87) [inline keyboards/jukaie/jk01/keymaps/via/keymap.c:23]:
```suggestion
    KC_MC = QK_USER,
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/jukaie/jk01/keymaps/via/keymap.c [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+    writePin(C15, keymap_config.no_gui);
+};
+
+void keyboard_pre_init_user(void) {
+
+    setPinInputLow(C15);
+    writePin(C15, false);
+    setPinInputLow(C0);
+    writePin(C0, false);
+};
```
Comment (drashna) [inline keyboards/jukaie/jk01/keymaps/via/keymap.c:87]: These should probably be moved to the keyboard level.

File: keyboards/jukaie/jk01/keymaps/via/keymap.c [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+  KC_LSFT,  KC_Z,     KC_X,     KC_C,     KC_V,     KC_B,     KC_N,     TO(Mac),  KC_COMM,  KC_DOT,   KC_SLSH,  KC_RSFT,            RGB_VAI,  KC_PGDN,
+  KC_LCTL,  KC_LALT,  KC_LGUI,                      KC_SPC,                                 KC_RALT,  MO(Winfn),KC_RCTL,  RGB_SPD,  RGB_VAD,  RGB_SPI),
+
+};
+
+void led_init_ports(void) {
+  setPinOutput(C0);
+  setPinOutput(C14);
+  setPinOutput(C15);
+};
```
Comment (drashna) [inline keyboards/jukaie/jk01/keymaps/via/keymap.c:137]: This is redundant.  Should be called in only one place.

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/jukaie/jk01/info.json [hunk 9]
```diff
@@ -0,0 +1,260 @@
...
+        "mousekey": true,
+        "nkro": true,
+        "rgb_matrix": true,
+        "rgblight": false
+    },
+    "indicators": {
+        "caps_lock": "C14"
+    },
+    "keyboard_folder": "jukaie/jk01",
+    "keymaps": {},
```
Comment (drashna) [inline keyboards/jukaie/jk01/info.json:50]:
```suggestion
```

File: keyboards/jukaie/jk01/info.json [hunk 10]
```diff
@@ -0,0 +1,260 @@
...
+    },
+    "parse_errors": [],
+    "parse_warnings": [],
+    "platform_key": "chibios",
+    "processor": "WB32FQ95",
+    "processor_type": "arm",
+    "protocol": "ChibiOS",
+    "qmk": {
+        "tap_keycode_delay": 10
+    },
```
Comment (drashna) [inline keyboards/jukaie/jk01/info.json:70]:
```suggestion
    "processor": "WB32FQ95",
    "qmk": {
        "tap_keycode_delay": 10
    },
```

File: keyboards/jukaie/jk01/info.json [hunk 11]
```diff
@@ -0,0 +1,260 @@
...
+    "usb": {
+        "device_version": "0.0.2",
+        "pid": "0x0002",
+        "vid": "0x0191"
+    },
+    "layouts": {
+        "LAYOUT": {
+            "c_macro": true,
+            "filename": "keyboards/jukaie/jk01/jk01.h",
+            "json_layout": true,
```
Comment (drashna) [inline keyboards/jukaie/jk01/info.json:172]:
```suggestion
```

File: keyboards/jukaie/jk01/jk01.c [hunk 12]
```diff
@@ -0,0 +1,172 @@
...
+#endif
+
+void keyboard_pre_init_user(void) {
+    setPinOutput(C0);
+    setPinOutput(C15);
+};
+
+void housekeeping_task_user(void) {
+    writePin(C15, keymap_config.no_gui);
+};
```
Comment (drashna) [inline keyboards/jukaie/jk01/jk01.c:172]:
```suggestion
void keyboard_pre_init_kb(void) {
    setPinOutput(C0);
    setPinOutput(C15);
    keyboard_pre_init_user();
};

void housekeeping_task_kb(void) {
    writePin(C15, keymap_config.no_gui);
};
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by jukaie** [comment]
@drashna @waffle87 resolved requested changes, please take a look at this PR, thanks.

File: keyboards/jukaie/jk01/keymaps/default/keymap.c [hunk 13]
```diff
@@ -0,0 +1,134 @@
...
+
+    switch (keycode) {
+
+        case KC_MC:
+            if (record->event.pressed) {
+                host_consumer_send(0x29F);
+            } else {
+                host_consumer_send(0);
+            }
+            return false; /* Skip all further processing of this key */
```
Comment (drashna) [inline keyboards/jukaie/jk01/keymaps/default/keymap.c:40]: This functionality is already in core, as `KC_MISSION_CONTROL`
https://docs.qmk.fm/#/keycodes_basic?id=media-keys

File: keyboards/jukaie/jk01/info.json [hunk 14]
```diff
@@ -0,0 +1,228 @@
...
+    "url": "",
+    "usb": {
+        "device_version": "0.0.2",
+        "pid": "0x0002",
+        "vid": "0x0191"
+    },
+    "layouts": {
+        "LAYOUT": {
+            "layout": [
+                { "label": "ESC", "matrix": [1, 3], "w": 1, "x": 0, "y": 0 },
```
Comment (fauxpark) [inline keyboards/jukaie/jk01/info.json:142]:
```suggestion
                { "label": "ESC", "matrix": [1, 3], "x": 0, "y": 0 },
```
etc...

File: keyboards/jukaie/jk01/info.json [hunk 15]
```diff
@@ -0,0 +1,228 @@
...
+    "bootmagic": {
+        "matrix": [1, 3]
+    },
+    "diode_direction": "COL2ROW",
+    "encoder": {
+        "rotary": [
+            {
+                "pin_a": "C2",
+                "pin_b": "C3"
+            }
```
Comment (fauxpark) [inline keyboards/jukaie/jk01/info.json:15]:
```suggestion
            {"pin_a": "C2", "pin_b": "C3"}
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/jukaie/jk01/info.json [hunk 16]
```diff
@@ -0,0 +1,225 @@
...
+            { "flags": 4, "matrix": [0, 5], "x": 150, "y": 50 },
+            { "flags": 4, "matrix": [11, 0], "x": 0, "y": 1 },
+            { "flags": 4, "matrix": [11, 1], "x": 0, "y": 2 }
+        ]
+    },
+    "url": "",
+    "usb": {
+        "device_version": "0.0.2",
+        "pid": "0x0002",
+        "vid": "0x0191"
```
Comment (drashna) [inline keyboards/jukaie/jk01/info.json:134]: While this VID wasn't in use, it's very low in the assigned ranges, and in a spot that is unused.  That makes me uncomfortable.  

Ideally, the higher this value is, the better.  And the further above 0x4000, the better.

**Comment by jukaie** [comment]
@drashna @waffle87 resolved requested changes, please take a look at this PR, thanks.

File: keyboards/jukaie/jk01/jk01.c [hunk 17]
```diff
@@ -0,0 +1,172 @@
...
+    if (index == 0) {
+        if (clockwise) {
+            tap_code(KC_VOLD);
+        } else {
+            tap_code(KC_VOLU);
+        }
+    }
+    return true;
+}
+#endif
```
Comment (zvecr) [inline keyboards/jukaie/jk01/jk01.c:163]: This is the default behaviour now
```suggestion
```

File: keyboards/jukaie/jk01/rules.mk [hunk 18]
```diff
@@ -0,0 +1,6 @@
+# Build Options
+#   change yes to no to disable
+#
+
```
Comment (zvecr) [inline keyboards/jukaie/jk01/rules.mk:4]:
```suggestion
```

File: keyboards/jukaie/jk01/info.json [hunk 19]
```diff
@@ -0,0 +1,225 @@
...
+        "backlight": false,
+        "bootmagic": true,
+        "command": false,
+        "console": false,
+        "encoder": true,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true,
+        "rgb_matrix": true,
+        "rgblight": false
```
Comment (zvecr) [inline keyboards/jukaie/jk01/info.json:26]:
```suggestion
        "bootmagic": true,
        "command": false,
        "console": false,
        "encoder": true,
        "extrakey": true,
        "mousekey": true,
        "nkro": true,
        "rgb_matrix": true
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/jukaie/jk01/keymaps/default/keymap.c [hunk 20]
```diff
@@ -0,0 +1,125 @@
...
+  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  RGB_HUI,
+  KC_TRNS,  KC_TRNS,  TO(Win),  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,
+  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,            KC_TRNS,  KC_TRNS,
+  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  TO(Mac),  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,            RGB_VAI,  KC_TRNS,
+  KC_TRNS,  KC_TRNS,  KC_TRNS,                      KC_TRNS,                                KC_TRNS,  KC_TRNS,  KC_TRNS,  RGB_SPD,  RGB_VAD,  RGB_SPI),
+
+};
+
+#if defined(ENCODER_MAP_ENABLE)
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][2] = {
```
Comment (zvecr) [inline keyboards/jukaie/jk01/keymaps/default/keymap.c:114]:
```suggestion
const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
```

File: keyboards/jukaie/jk01/keymaps/via/keymap.c [hunk 21]
```diff
@@ -0,0 +1,125 @@
...
+  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  RGB_HUI,
+  KC_TRNS,  KC_TRNS,  TO(Win),  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,
+  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,            KC_TRNS,  KC_TRNS,
+  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  TO(Mac),  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,            RGB_VAI,  KC_TRNS,
+  KC_TRNS,  KC_TRNS,  KC_TRNS,                      KC_TRNS,                                KC_TRNS,  KC_TRNS,  KC_TRNS,  RGB_SPD,  RGB_VAD,  RGB_SPI),
+
+};
+
+#if defined(ENCODER_MAP_ENABLE)
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][2] = {
```
Comment (zvecr) [inline keyboards/jukaie/jk01/keymaps/via/keymap.c:114]:
```suggestion
const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
```

**Review by zvecr — CHANGES_REQUESTED** [review]

**Comment by jukaie** [comment]
@drashna @fauxpark @zvecr @waffle87 resolved requested changes. Can you please take a look at this PR, thanks.

File: keyboards/jukaie/jk01/info.json [hunk 22]
```diff
@@ -0,0 +1,222 @@
...
+    },
+    "matrix_pins": {
+        "cols": ["A0", "A1", "A2", "A3", "A4", "A8", "A9", "A10"],
+        "rows": ["B0", "B1", "B2", "B3", "B4", "B5", "B6", "B7", "B8", "B9", "B10", "B12"]
+    },
+    "processor": "WB32FQ95",
+    "qmk": {
+        "tap_keycode_delay": 10
+    },
+    "rgb_matrix": {
```
Comment (dunk2k) [inline keyboards/jukaie/jk01/info.json:36]:
```suggestion
    "rgb_matrix": {
        "animations": {
            "solid_color": true,
            "alphas_mods": true,
            "gradient_up_down": true,
            "gradient_left_right": true,
            "breathing": true,
            "band_sat": true,
            "band_val": true,
            "band_pinwheel_sat": true,
            "band_pinwheel_val": true,
            "band_spiral_sat": true,
            "band_spiral_val": true,
            "cycle_all": true,
            "cycle_left_right": true,
            "cycle_up_down": true,
            "cycle_out_in": true,
            "cycle_out_in_dual": true,
            "rainbow_moving_chevron": true,
            "cycle_pinwheel": true,
            "cycle_spiral": true,
            "dual_beacon": true,
            "rainbow_beacon": true,
            "rainbow_pinwheels": true,
            "raindrops": true,
            "jellybean_raindrops": true,
            "hue_breathing": true,
            "hue_pendulum": true,
            "hue_wave": true,
            "pixel_fractal": true,
            "pixel_flow": true,
            "pixel_rain": true,
            "typing_heatmap": true,
            "digital_rain": true,
            "solid_reactive_simple": true,
            "solid_reactive": true,
            "solid_reactive_wide": true,
            "solid_reactive_multiwide": true,
            "solid_reactive_cross": true,
            "solid_reactive_multicross": true,
            "solid_reactive_nexus": true,
            "solid_reactive_multinexus": true,
            "splash": true,
            "multisplash": true,
            "solid_splash": true,
            "solid_multisplash": true
        },
```

File: keyboards/jukaie/jk01/config.h [hunk 23]
```diff
@@ -0,0 +1,93 @@
...
+#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_WIDE
+#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTIWIDE
+#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_CROSS
+#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTICROSS
+#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_NEXUS
+#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTINEXUS
+#define ENABLE_RGB_MATRIX_SPLASH
+#define ENABLE_RGB_MATRIX_MULTISPLASH
+#define ENABLE_RGB_MATRIX_SOLID_SPLASH
+#define ENABLE_RGB_MATRIX_SOLID_MULTISPLASH
```
Comment (dunk2k) [inline keyboards/jukaie/jk01/config.h:93]: move to `info.json`
```suggestion
```

**Review by dunk2k — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Comment by jukaie** [comment]
@fauxpark @waffle87 @zvecr can you please take a look at this PR, thanks.

File: keyboards/jukaie/jk01/keymaps/via/keymap.c [hunk 24]
```diff
@@ -0,0 +1,125 @@
...
+            }
+            return false; /* Skip all further processing of this key */
+
+        case KC_EJ:
+            if (record->event.pressed) {
+                host_consumer_send(0x0B8);
+            } else {
+                host_consumer_send(0);
+            }
+            return false; /* Skip all further processing of this key */
```
Comment (waffle87) [inline keyboards/jukaie/jk01/keymaps/via/keymap.c:63]:
```suggestion
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

**Comment by jukaie** [comment]
@waffle87 @fauxpark @zvecr resolved requested changes. Could you please take a look at this PR, thanks.

**Comment by jukaie** [comment]
@waffle87 @fauxpark @zvecr Can someone please take a look at this PR, thanks.

File: keyboards/jukaie/jk01/info.json [hunk 25]
```diff
@@ -0,0 +1,267 @@
...
+            "solid_reactive_multicross": true,
+            "solid_reactive_nexus": true,
+            "solid_reactive_multinexus": true,
+            "splash": true,
+            "multisplash": true,
+            "solid_splash": true,
+            "solid_multisplash": true
+        },
+        "center_point": [76, 25],
+        "driver": "AW20216",
```
Comment (fauxpark) [inline keyboards/jukaie/jk01/info.json:83]:
```suggestion
        "driver": "aw20216",
```

File: keyboards/jukaie/jk01/info.json
Comment (fauxpark) [inline keyboards/jukaie/jk01/info.json:1]: Please run `qmk format-json` on this file.

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by noroadsleft — APPROVED** [review]
Looks good to me.

**Comment by noroadsleft** [comment]
For future reference, we recommend against committing to your `master` branch as you've done here, because pull requests from modified `master` branches can make it more difficult to keep your QMK fork updated. It is highly recommended for QMK development – regardless of what is being done or where – to keep your master updated, but **NEVER** commit to it. Instead, do all your changes in a branch (branches are basically free in Git) and issue PRs from your branches when you're developing.

There are instructions on how to keep your fork updated here:

[**Best Practices: Your Fork's Master: Update Often, Commit Never**](https://docs.qmk.fm/#/newbs_git_using_your_master_branch)

[Fixing Your Branch](https://docs.qmk.fm/#/newbs_git_resynchronize_a_branch) will walk you through fixing up your `master` branch moving forward. If you need any help with this just ask.

Thanks for contributing!

**Pull request merged**